### ISSUE TITLE
Rule Type "Variable Metadata Check against Library Metadata" doesn't work

### ISSUE THREAD

**Issue by HajimeShimizu** [cdisc-org/cdisc-rules-engine#1261]
When rule type "Variable Metadata Check against Library Metadata" is selected, error is reported.

```
[ERROR 2025-07-23 14:28:12,416 - console_logger.py:58] - Error occurred during validation.
            Error: 'dataset_name'
            Error Type: <class 'KeyError'>
            Error Message: 'dataset_name'
```

As far as I see, this error is triggered at def get_dataset_contents in base_dataset_builder.py
```
def get_dataset_contents(self, **kwargs):
  if ....
  else:
     dataset: DatasetInterface = self.data_service.get_dataset(self.dataset_path)  # <- Error
```

if this part is updated as follows, it works.
```
   dataset: DatasetInterface = self.data_service.get_dataset(self.dataset_path,
                                                                      dataset_name=self.dataset_path,
                                                                      **kwargs)
```

However, this bug fix will guide us to new findings in variable_metadata_with_library_metadata.py
```
data = content_variables_metadata.merge(
            library_variables_metadata.data,
            how="left",
            left_on="variable_name",
            right_on="library_variable_name",
        ).fillna("")
```

this fires KeyError: "library_variable_name"

I checked the columns in library_variable_metadata.data and found 'library_variable_name'.
If we choose several columns, merge process completed. Thus, any column(s) is not suitable data merge.

```
data = content_variables_metadata.merge(
            library_variables_metadata.data[['library_variable_name', 
                                             'library_variable_core',
                                             'library_variable_label']],
            how="left",
            left_on="variable_name",
            right_on="library_variable_name",
        ).fillna("")
```
